.github/workflows/ci.yml

```
@@ -18,24 +18,35 @@ jobs:
         uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
-      - name: Cache node modules
-        uses: actions/cache@v3
-        env:
-          cache-name: cache-node-modules
-        with:
-          path: node_modules
-          key: ${{ runner.os }}-node${{ runner.node-version }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
       - run: npm ci
       - run: npm run build
       - run: npm test
+      # Upload coverage for sonarcube
+      - uses: actions/upload-artifact@v3
+        with:
+          name: code-coverage
+          path: coverage/
 
+  sonar-scan:
+    needs: [unit-tests]
+    runs-on: ${{ matrix.os }}
+    if: ${{ github.actor != 'dependabot[bot]' }}
+    strategy:
+      matrix:
+        # Available OS's: https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners
+        os: [ubuntu-latest]
+        node-version: [16.x]
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/download-artifact@v3
+        with:
+          name: code-coverage
+          path: coverage/
       # Sonar analysis needs the full history for features like automatic assignment of bugs. If the following step
       # is not included the project will show a warning about incomplete information.
       - run: git fetch --unshallow
```

```suggestion
      # Sonar analysis needs the full history for features like automatic assignment of bugs hence fetch-depth: 0.
      - uses: actions/checkout@v3
         with:
           fetch-depth: 0
      - uses: actions/download-artifact@v3
        with:
          name: code-coverage
          path: coverage/
```
This makes the initial checkout do the necessary full fetch instead of doing it in a subsequent step

---

e173a059911afe4b7a446472fd672fa9fd4ac322